tests/bluetooth/bsim/mesh/src/test_dfu.c

```diff
@@ -511,6 +511,164 @@ static void test_dist_dfu_self_update(void)
 	PASS();
 }
 
+static void test_dist_dfu_slot_create(void)
+{
+	const struct bt_mesh_dfu_slot *slot[3];
+	size_t size = 100;
+	uint8_t fwid[CONFIG_BT_MESH_DFU_FWID_MAXLEN] = { 0 };
+	size_t fwid_len = 4;
+	uint8_t metadata[CONFIG_BT_MESH_DFU_METADATA_MAXLEN] = { 0 };
+	size_t metadata_len = 4;
+	const char *uri = "test";
+	int err, i;
+
+	settings_test_backend_clear();
+
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &dist_comp);
+	dist_prov_and_conf(DIST_ADDR);
+
+	for (i = 0; i < CONFIG_BT_MESH_DFU_SLOT_CNT; i++) {
```

Comment:
If the test is dependent on `CONFIG_BT_MESH_DFU_SLOT_CNT` being set to some spesific value, it vould be nice to have a assert check that verifies that it is set correct. Since the configuration is shared by many tests it might happen that someone for reasons want to change the option. Having a guard like this with a meaningful message might make it easier to catch such issues.